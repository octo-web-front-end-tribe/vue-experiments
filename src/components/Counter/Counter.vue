<template>
  <div class="test">
    <h1>counter</h1>
    <p>{{ total }}</p>
    <button-counter @increment="incrementTotal"></button-counter>
    <button-counter @increment="incrementTotal"></button-counter>
  </div>
</template>

<script>
  import ButtonCounter from '../ButtonCounter/ButtonCounter'

  export default {
    data () {
      return {
        total: 0
      }
    },
    methods: {
      incrementTotal: function () {
        this.total += 1
      }
    },
    components: {
      ButtonCounter
    }
  }
</script>

<style scoped>
  .test {
    font-size: 40px;
  }
</style>
