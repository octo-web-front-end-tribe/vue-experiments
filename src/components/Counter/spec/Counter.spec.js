import {shallow} from 'vue-test-utils';
import Counter from '../Counter';

describe('Counter.vue', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallow(Counter);

    // then
    expect(wrapper.element).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should initialize counter to 0', () => {
    // when
    const wrapper = shallow(Counter);

    // then
    let paragraph = wrapper.find('p');
    expect(paragraph.text()).toEqual('0');
  });

  it('should increment counter on event increment', () => {
    // given
    const wrapper = shallow(Counter);

    // when
    wrapper.vm.incrementTotal();
    wrapper.update();

    // then
    let paragraph = wrapper.find('p');
    expect(paragraph.text()).toEqual('1');
  });
});
