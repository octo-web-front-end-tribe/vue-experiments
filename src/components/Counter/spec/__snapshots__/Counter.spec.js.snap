// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue should match snapshot 1`] = `
<div
  class="test"
>
  <h1>
    counter
  </h1>
   
  <p>
    0
  </p>
   
  <!---->
   
  <!---->
</div>
`;
